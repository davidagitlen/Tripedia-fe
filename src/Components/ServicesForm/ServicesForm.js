import React, { useState, useContext } from 'react';
import './ServicesForm.scss';
import { FormContext } from '../../Contexts/FormContext';

const ServicesForm = ({ collapseForm, openForm, defaultForm }) => {
  const { formState, setFormState } = useContext(FormContext);
  console.log('servicesForm', formState);

  const [form, toggleClicked] = useState({
    GasStation: false,
    Hospital: false,
    Mechanic: false,
    GroceryStore: false,
    RestStop: false
  });

  let mockProps = [
    "Gas Station",
    "Hospital",
    "Mechanic",
    "Grocery Store",
    "Rest Stop"
  ];

  const checkBoxes = mockProps.map(checkBox => {
    let name = checkBox.replace(/ /gi, "");
    return (
      <div key={name} className="individual-checkbox__container">
        <input
          className="checkbox"
          type="checkbox"
          value={name}
          checked={form[name]}
          onChange={e =>
            toggleClicked({
              ...form,
              [e.target.value]: !form[e.target.value]
            })
          }
        />
        <label>{checkBox}</label>
      </div>
    );
  });
  if (openForm.ServicesForm) {
    return (
      <div className="form__container">
        <p>- Services -</p>
        <div className="checkbox__container">{checkBoxes}</div>
      </div>
    );
  } else {
    return (
      <div
        className="form-closed__container"
        onClick={e =>
          collapseForm({
            ...defaultForm,
            ServicesForm: !openForm.ServicesForm
          })
        }
      >
        <p>- ServicesForm -</p>
      </div>
    );
  }
};

export default ServicesForm;
