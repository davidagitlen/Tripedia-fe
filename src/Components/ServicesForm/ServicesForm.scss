.services-form__container {
  width: 100%;
  height: 32vh;
  background: blueviolet;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.services-form-closed__container {
  width: 100%;
  height: 8vh;
  background: orange;
}

.food__checkbox {
  margin: 0px 10px 0px 0px;
}