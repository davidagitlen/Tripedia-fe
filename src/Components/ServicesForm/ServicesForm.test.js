import React from 'react';
import { shallow } from 'enzyme';
import ServicesForm from './ServicesForm';

describe('ServicesForm', () => {
  
  it("should pass a dummy test", () => {
    expect(true).toEqual(true);
  });
});