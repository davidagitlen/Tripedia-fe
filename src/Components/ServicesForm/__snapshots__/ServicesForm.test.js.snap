// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesForm should mock formContext and match a snapshot 1`] = `
<ServicesForm
  collapseForm={[MockFunction]}
  defaultForm={
    Object {
      "AccommodationsForm": false,
      "AttractionsForm": false,
      "DrinksForm": false,
      "FoodForm": false,
      "ServicesForm": false,
      "StartForm": false,
      "SubmitTripForm": false,
    }
  }
  formObject={
    Object {
      "ArtGalleries": Array [],
      "DayCamps": Array [],
      "WalkingTours": Array [],
    }
  }
  openForm={[MockFunction]}
>
  <div
    className="form-closed__container"
    onClick={[Function]}
  >
    <img
      alt="services"
      src="services-pin.svg"
    />
    <p>
      - ServicesForm -
    </p>
  </div>
</ServicesForm>
`;
